suricata/update/main.py

```diff
@@ -353,6 +353,16 @@ def fetch(self, url):
         net_arg = url
         url = url[0] if isinstance(url, tuple) else url
         tmp_filename = self.get_tmp_filename(url)
+        if config.args().offline:
+            if config.args().force:
+                logger.warning("Failed to force remote rule files to be "
+                               "downloaded.")
+            logger.info("Using latest cached version of source %s", url)
+            if not os.path.exists(tmp_filename):
+                logger.error("Can't proceed offline, "
+                             "source %s has not yet been downloaded.", url)
+                sys.exit(1)
+            return self.extract_files(tmp_filename)
```

Comment:
Hmm all of these are logged per source. So, using `--offline` with `-f` gives 
```
27/6/2019 -- 16:20:02 - <Warning> -- Failed to force remote rule files to be downloaded.
27/6/2019 -- 16:20:02 - <Info> -- Using latest cached version of source https://sslbl.abuse.ch/blacklist/sslblacklist.rules
27/6/2019 -- 16:20:02 - <Warning> -- Failed to force remote rule files to be downloaded.
27/6/2019 -- 16:20:02 - <Info> -- Using latest cached version of source https://security.etnetera.cz/feeds/etn_aggressive.rules
27/6/2019 -- 16:20:02 - <Warning> -- Failed to force remote rule files to be downloaded.
27/6/2019 -- 16:20:02 - <Info> -- Using latest cached version of source https://rules.emergingthreats.net/open/suricata-5.0.0/emerging.rules.tar.gz
```
for three enabled sources.

Reply:
@shivan1b Check for `-f` and `--offline` should be added in fetch function only? Or is it ok to check before this function is called per source? Something like this:
```
if config.args().offline:
    if config.args().force:
        logger.warning("Failed to force rule files to be downloaded")
    logger.info("Proceeding offline")

# Now download each URL.
for url in urls:
    Fetch().run(url, files)
```